### DISCUSSION TITLE
I wan to use minio  alternative s3

### DISCUSSION THREAD

**Discussion by magic-thomas** [adonisjs/core#3868] [Help] [answered]
hi 
I wan to use private VPS with minio package instead s3. 

I have storage server IP , minio access key,  minio scret key only. 

How can I change the driver.ts for minio package ? 

// driver.ts 

```js
    s3: {
        driver: 's3',
        visibility: 'private',
        key: Env.get('AWS_ACCESS_KEY'),
        secret: Env.get('AWS_SECRET_KEY'),
        region: Env.get('S3_REGION'),
        bucket: Env.get('S3_BUCKET'),
    },
```

**Comment by Julien-R44** [comment, marked as answer]
The only thing you are missing, to make MinIO work, is this property: 
```ts
...
  s3: {
      forcePathStyle: true, // 👈 Don't forget this !
      key: Env.get('S3_KEY'),
      secret: Env.get('S3_SECRET'),
      region: Env.get('S3_REGION'),
      bucket: Env.get('S3_BUCKET'),
      endpoint: Env.get('S3_ENDPOINT'),
  }
...
```

**Reply by magic-thomas** [reply]
thanks for reply.. 


I guess 'S3_ENDPOINT' is the minIO server address.

but what is that for  'S3_REGION'  ?

**Reply by Julien-R44** [reply]
It is possible to customize the region of your MinIO server from  Settings > Region dashboard. 
![image](https://user-images.githubusercontent.com/8337858/185350211-9b9b694e-3fed-48e6-89f6-461afd1e4d7f.png)

I'm not sure, but I guess `us-east-1` is the default value. In my case, I have not defined the region, use `S3_REGION=us-east-1` and it works.

**Reply by magic-thomas** [reply]
I am successful with this. 

```js
s3: {
    driver: 's3',
    visibility: 'public',
    forcePathStyle: true, 
    key: 'key',
    secret: 'secret',
    region: 'my-city',
    bucket: 'buck01',
    endpoint: 'http://my-minio-ip:9000'
},

```

Thanks.
